patroni/postgresql/slots.py

```diff
@@ -417,8 +417,24 @@ def _ensure_physical_slots(self, slots: Dict[str, Any], clean_inactive_physical_
             # And advance restart_lsn on physical replication slots that are not expected to be active.
             elif self._postgresql.can_advance_slots and self._replication_slots[name]['type'] == 'physical' and\
                     value.get('expected_active') is not True and not value['xmin']:
+                restart_lsn = value.get('restart_lsn')
+                if not restart_lsn:
+                    # This slot either belongs to a member or was configured as a permanent slot. However, for some
+                    # reason the slot was created by an external agent instead of by Patroni, and it was created without
+                    # reserving the LSN. We drop the slot here, as we cannot advance it, and let Patroni recreate and
+                    # manage it in the next cycle.
+                    logger.warning('Dropping physical replication slot %s because it has no restart_lsn. '
+                                   'This slot was probably not created by Patroni, but by an external agent.',
+                                   name)
+                    _, dropped = self.drop_replication_slot(name)
+                    if dropped:
+                        self._replication_slots.pop(name)
+                    else:
+                        self._schedule_load_slots = True
+                        logger.error("Failed to drop replication slot '%s'", name)
+                    continue
```

Comment:
These lines are not covered by tests:
```python
tests/test_slots.py ..
>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>
> /home/akukushkin/git/patroni/patroni/postgresql/slots.py(431)_ensure_physical_slots()
-> _, dropped = self.drop_replication_slot(name)
(Pdb) n
ValueError: not enough values to unpack (expected 2, got 0)
> /home/akukushkin/git/patroni/patroni/postgresql/slots.py(431)_ensure_physical_slots()
-> _, dropped = self.drop_replication_slot(name)
```

Actually, this block is almost the same as https://github.com/patroni/patroni/blob/bc0c4234fedb5f4268b1f089208286b971e88ad2/patroni/postgresql/slots.py#L397-L405
I think we need to move it to a dedicated method `_drop_physical_slot()`.

Reply:
I just sent a couple of commits:

* 1611d1d71f4b02e4f4e7fcea5f42337fe5e96516: abstracting the common logic in a new helper method `_drop_physical_slot`
* 0beeb12cce9f8eec3a5d677cf247d12f5f45a452: to modify the unit test suite. I left a note in that commit message, about a couple of patches to `drop_replication_slot` which I didn't modify yet. It seems to me those unit tests "will have no value" if I replace the mock, so we might want either to remove them or to keep as it is.